packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -122,14 +122,63 @@ function getDocumentIdStrategy(type?: TestDriverTypes): IDocumentIdStrategy {
     }
 }
 
+export class ErrorTrackingLogger extends TelemetryLogger{
```

Comment:
Can you please add a brief description of what this logger does. It's not very apparent what this is used for.